app/controllers/subject_controller.rb

```diff
@@ -50,11 +43,19 @@ class App < Sinatra::Application
   delete_subject = lambda do
     @subject = Subject.find_by(id: params[:id])
     @subject.delete
+    redirect '/categories'
+  end
+
+  delete_subject = lambda do
+    @subject = Subject.find_by(id: params[:id])
+    @subject.delete
+    redirect '/categories'
   end
 
-  get '/categories', &get_categories
-  get '/addcategories', &add_categories_page
-  post '/categories', &create_subject
-  post '/categories/add', &add_categories
-  delete '/categories/:id', &delete_subject
+  get '/categories', auth: true, &get_categories
```

Comment:
Let's have the links at the top of their controller methods. I understand that lambdas are extra knowledge but I don't think they are necessary in this context